internal/httputil/pagination.go

```diff
@@ -0,0 +1,58 @@
+package httputil
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+)
+
+const maxPageSize = 100
+
+type PaginationData struct {
+	PageNumber int
+	PageSize   int
+	SortBy     map[string]string
+	FilterBy   map[string]string
+}
+
+func ParsePaginationQuery(r *http.Request) (*PaginationData, error) {
+	qs := r.URL.Query()
+
+	pagination := &PaginationData{
+		PageNumber: 0,
+		PageSize:   10,
+	}
+
+	pageNumber := qs.Get("page")
+	if pageNumber != "" {
+		pn, err := strconv.Atoi(pageNumber)
+
+		if pn < 0 || err != nil {
+			return nil, errors.New("invalid page number. value should not be less than 0")
+		}
+
+		pagination.PageNumber = pn
+	}
+
+	pageSize := qs.Get("size")
+	if pageSize != "" {
```

Comment:
This if clause is redundant. You get an error from strconv.Atoi if you pass empty string into it.

Reply:
this if clause makes pageSize query parameter required, I'm not sure if that's the proper behaviour